### ISSUE TITLE
Implement validation loss during training

### ISSUE THREAD

**Issue by sfmig** [SainsburyWellcomeCentre/crabs-exploration#159]
Logging validation loss is trickier than expected due to the model we are using.

- We are using `fasterrcnn_resnet50_fpn_v2` that returns [predictions only](https://pytorch.org/vision/main/models/generated/torchvision.models.detection.fasterrcnn_resnet50_fpn.html#torchvision.models.detection.fasterrcnn_resnet50_fpn) if set to eval
- What can we do?
	- redefine [`eager_outputs()`](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/generalized_rcnn.py#L118) method of the parent class `GeneralizedRCNN`? 
		- not sufficient, even if we hack this method the loss is still an empty dict when run in validation - probably we need to do something like [this](https://stackoverflow.com/a/72321898)
	- use a custom `forward` method?
		- [ptrblk to the rescue](https://discuss.pytorch.org/t/how-to-retrieve-the-loss-function-of-fasterrcnn-for-object-detection/174100)
		- probably this is our best bet, but I think it's more difficult than what Patrick suggests because all of the stages (`rpn`, `roi_heads`)  behave the same as the "parent" network: they only return losses during training, `detector_losses` and `proposal_losses` are empty dicts if in eval mode.
		- [this person reimplemented all the bits](https://stackoverflow.com/questions/71288513/how-can-i-determine-validation-loss-for-faster-rcnn-pytorch) in an old version of the codebase, probably we can replicate (preferably without hacking torchvision directly).
    - use another model?
